producer.go

```diff
@@ -369,7 +402,8 @@ func (w *Producer) transactionCleanup() {
 }
 
 func (w *Producer) log(lvl LogLevel, line string, args ...interface{}) {
-	logger, logLvl := w.getLogger()
+	logger := w.getLogger(lvl)
+	logLvl := w.getLogLevel()
```

Comment:
calling two methods here ends up doing RLock and RUnlock twice ... could still be a single function `getLogger()` that also returns current level

Reply:
I prefer the separation but you make a good point.